nebula_decoders/include/nebula_decoders/nebula_decoders_hesai/decoders/hesai_decoder.hpp

```diff
@@ -76,13 +123,32 @@ class HesaiDecoder : public HesaiScanDecoder
 
   rclcpp::Logger logger_;
 
+  // filtered pointcloud counter
+  HesaiDecodeFilteredInfo decode_filtered_info_;
+
   /// @brief For each channel, its firing offset relative to the block in nanoseconds
   std::array<int, SensorT::packet_t::n_channels> channel_firing_offset_ns_;
   /// @brief For each return mode, the firing offset of each block relative to its packet in
   /// nanoseconds
   std::array<std::array<int, SensorT::packet_t::n_blocks>, SensorT::packet_t::max_returns>
     block_firing_offset_ns_;
 
+  void get_minmax_info(const NebulaPoint & point)
```

Comment:
This can be moved into `HesaiDecodeFilteredInfo` so it is clear to readers that it belongs to that struct.